code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -503,27 +500,16 @@ generate/load female uniform sprites matching all previously decided variables
 				t_state = inhand_icon_state
 		else if(worn_icon_state)
 			t_state = worn_icon_state
-	var/t_icon = worn_icon
-	if(!t_icon)
-		t_icon = default_icon_file
 
 	//Find a valid icon file from variables+arguments
-	var/file2use
-	if(!isinhands && worn_icon)
-		file2use = worn_icon
-	if(!file2use)
-		file2use = default_icon_file
+	var/file2use = !isinhands ? (worn_icon ? worn_icon : default_icon_file) : default_icon_file
 
 	//Find a valid layer from variables+arguments
-	var/layer2use
-	if(alternate_worn_layer)
-		layer2use = alternate_worn_layer
-	if(!layer2use)
-		layer2use = default_layer
+	var/layer2use = alternate_worn_layer ? alternate_worn_layer : default_layer
 
 	var/mutable_appearance/standing
 	if(femaleuniform)
-		standing = wear_female_version(t_state,file2use,layer2use,femaleuniform)
+		standing = wear_female_version(t_state,file2use,layer2use,femaleuniform) //should layer2use be in sync with the adjusted value below? needs testing - shiz
 	if(!standing)
 		standing = mutable_appearance(file2use, t_state, -layer2use)

```

Comment:
this is -layer2use, but females use just layer2use. have a feeling this should probably be in sync.